core/blockchain_action.go

```diff
@@ -358,6 +358,14 @@ func (chain *FullBlockChain) addBlockOnChain(source string, b *types.Block) (ret
 }
 
 func (chain *FullBlockChain) transitAndCommit(block *types.Block) (ok bool, err error) {
+
+	// Check if all txs exist in the pool to ensure the basic validation is done
+	for _, tx := range block.Transactions {
+		if exist, _ := chain.GetTransactionPool().IsTransactionExisted(tx.Hash); !exist {
+			return false, fmt.Errorf("tx is not in the pool %v", tx.Hash.Hex())
+		}
+	}
+
```

Comment:
why need to check all transactions in pool?

Reply:
ensure all the transactions are checked